modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1257,6 +1258,8 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                 model.imports.add("ArrayList");
             } else if ("set".equals(property.containerType)) {
                 model.imports.add("LinkedHashSet");
+                model.imports.add("JsonDeserialize");
+                property.vendorExtensions.put("x-setter-extra-annotation", "@JsonDeserialize(using = LinkedHashSet.class)");
```

Comment:
I have also tried the same in one of the scenarios. But gave me compilation errors. The workaround works as expected when you modify `@JsonDeserialize(using = LinkedHashSet.class)` to `@JsonDeserialize(as = LinkedHashSet.class)`.

Reply:
thanks @sonalidhome. fixed

Reply:
There is compilation error.  Can you please try to re-generate and push it? 
`[ERROR] COMPILATION ERROR : 
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2/src/main/java/org/openapitools/client/model/Pet.java:[18,49] package com.fasterxml.jackson.databind.annotation does not exist
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2/src/main/java/org/openapitools/client/model/Pet.java:[205,4] cannot find symbol
  symbol:   class JsonDeserialize
  location: class org.openapitools.client.model.Pet
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project petstore-retrofit2: Compilation failure: Compilation failure:
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2/src/main/java/org/openapitools/client/model/Pet.java:[18,49] package com.fasterxml.jackson.databind.annotation does not exist
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/retrofit2/src/main/java/org/openapitools/client/model/Pet.java:[205,4] cannot find symbol
[ERROR] symbol:   class JsonDeserialize
[ERROR] location: class org.openapitools.client.model.Pet
[ERROR] -> [Help 1]`

Reply:
@sonalidhome thanks. finally should be ok now